### PR TITLE
Copter: Move "land_run_horizontal_control()" to use position controller only

### PR DESCRIPTION
**Pull request by rishabsingh3003**
This uses the new Position Controller methods for landing instead of relying on loiter
Also removed direct injection of velocity into position controller

### CODE REVIEW

**Review by rishabsingh3003 — COMMENTED** [review]
The changes done to vertical control method should be completely NFC

File: ArduCopter/mode.cpp [hunk 1]
```diff
@@ -558,8 +558,14 @@ void Mode::land_run_vertical_control(bool pause_descent)
             const float precland_min_descent_speed = 10.0f;
 
             float max_descent_speed = abs(g.land_speed)*0.5f;
-            float land_slowdown = MAX(0.0f, pos_control->get_pos_error_xy_cm()*(max_descent_speed/precland_acceptable_error));
-            cmb_rate = MIN(-precland_min_descent_speed, -max_descent_speed+land_slowdown);
+
+            Vector2f target_pos;
+            if (copter.precland.get_target_position_cm(target_pos)) {
+                const Vector2f current_pos = inertial_nav.get_position().xy();
```
Comment (peterbarker) [inline ArduCopter/mode.cpp:564]:
```suggestion
                const Vector2f &current_pos = inertial_nav.get_position().xy();
```

It's dark magic.

File: libraries/AC_PrecLand/AC_PrecLand.cpp [hunk 2]
```diff
@@ -271,6 +278,31 @@ bool AC_PrecLand::get_target_velocity_relative_cms(Vector2f& ret)
...
 
+// get the absolute velocity of the vehicle
+void AC_PrecLand::get_target_velocity_cms(const Vector2f& vehicle_velocity_cms, Vector2f& target_vel_cms)
+{
+    if (!_moving) {
+        // the target should not be moving
+        target_vel_cms.zero();
+        return;
+    }
+    if ((EstimatorType)_estimator_type.get() == EstimatorType::RAW_SENSOR) {
```
Comment (peterbarker) [inline libraries/AC_PrecLand/AC_PrecLand.cpp:289]: Should use `AP_Enum<EstimatorType>` instead of `AP_Int8` as a type for `_estimator_type`

Pre-existing

**Comment by amilcarlucas** [comment]
This has a lot of conflicts. I backported it almost cleanly to Copter-4.1 but it needs a bigger rebase on master.
Do you plan to take this to the DevCall ?

**Comment by rishabsingh3003** [comment]
@amilcarlucas yes.. this breaks precision loiter I think because when I was testing on my copter it was misbehaving. I'll have to take a look.
The larger problem is that this changes how one repositions vehicle in land mode. It moves away from the loiter control, which is a huge change. Randy and others aren't convinced this is a good change yet. So I might have to tweak this PR to only keep the prec land parts and remove the changes to the reposition code..

**Comment by rishabsingh3003** [comment]
Closing in favour of #18584

**Comment by lthall** [comment]
Why did you remove the position controller change?

**Pull request closed without merging**